src/utils/body.ts

```diff
@@ -18,20 +19,37 @@ export type ParseBodyOptions = {
    * parseBody should return { file: ['aaa', 'bbb'], message: 'hello' }
    */
   all?: boolean
+  /**
+   * Determines whether all fields with dot notation should be parsed as nested objects.
+   * @default false
+   * @example
+   * const data = new FormData()
+   * data.append('obj.key1', 'value1')
+   * data.append('obj.key2', 'value2')
+   *
+   * If dot is false:
+   * parseBody should return { 'obj.key1': 'value1', 'obj.key2': 'value2' }
+   *
+   * If dot is true:
+   * parseBody should return { obj: { key1: 'value1', key2: 'value2' } }
+   */
+  dot?: boolean
 }
 
 export const parseBody = async <T extends BodyData = BodyData>(
   request: HonoRequest | Request,
-  options: ParseBodyOptions = { all: false }
+  options: ParseBodyOptions = {}
 ): Promise<T> => {
+  const { all = options?.all ?? false, dot = options?.dot ?? false } = options
```

Comment:
This could be just `const { all = false, dot = false } = options`

Reply:
This will override the values given in options to `false`

Reply:
If the value is not defined then it will override it to false.

Reply:
But anyway lets add that

Reply:
Added ✔️